src/main/java/com/example/weather/service/WeatherRequestServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package com.example.weather.service;
+
+import com.example.weather.kafka.KafkaProducer;
+import com.example.weather.model.Weather;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.Objects;
+@Service
+public class WeatherRequestServiceImpl implements WeatherRequestService {
+    @Autowired
+    private KafkaProducer kafkaProducer;
+
+    @Value("${weather.url}")
+    private String url;
+
+    @Value("${weather.apikey}")
+    private String apikey;
+
+    @Override
+    public void requestWeather(String city) throws JsonProcessingException {
```

Comment:
better change the name of this method, cause now it makes it unclear. if you request the weather, then why don't you return it? change to more appropriate name